### PR TITLE
feat: new component selectablecardgroup for ui and form

### PR DESCRIPTION
**Pull request by lisalupi**
## Type

- Feature

#### What is expected?

New component `SelectableCardGroup` (UI) and `SelectableCardGroupField`(form) to deal with group of SelectableCards.
#### The following changes were made:

(Describe what you did)

1. Reused the component `SelectableCard`to create a new component the same way `RadioGroup` and `Checkbox` group were done.

2. Tests & Storybook

### CODE REVIEW

File: packages/form/src/components/SelectableCardGroupField/index.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      | 'error'
+      | 'columns'
+      | 'children'
+      | 'showTick'
+      | 'type'
+      | 'className'
+      | 'onChange'
+    >
+  > &
+  Required<Pick<ComponentProps<typeof SelectableCardGroup>, 'legend'>>
```
Comment (johnrazeur) [inline packages/form/src/components/SelectableCardGroupField/index.tsx:27]:
```suggestion
 Pick<ComponentProps<typeof SelectableCardGroup>, 'legend'>
```

**Review by johnrazeur — APPROVED** [review]

File: .changeset/empty-islands-itch.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+"@ultraviolet/form": patch
+"@ultraviolet/ui": patch
+---
+
+New components `SelectableCardGroup` in ultraviolet/ui and `SelectableCardGroupField` in ultraviolet/form
```
Comment (matthprost) [inline .changeset/empty-islands-itch.md:6]: You can split the changeset in 2 files, one for ui and one for form with the correct component name for each

File: packages/ui/src/components/SelectableCardGroup/index.tsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  padding: 0;
+  margin: 0;
+`
+
+const StyledRequiredIcon = styled(Icon)`
+  vertical-align: super;
+`
+
+type SelectableCardGroupProps = {
+  legend: string
```
Comment (matthprost) [inline packages/ui/src/components/SelectableCardGroup/index.tsx:94]: We should make legend optional here cause we have plenty of use case without any legend, I know it is not how we did for radio and checkbox 😅

File: packages/ui/src/components/SelectableCardGroup/index.tsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    <SelectableCardGroupContext.Provider value={contextValue}>
+      <Stack gap={1}>
+        <FieldSet className={className}>
+          <Stack gap={1.5}>
+            <Text as="legend" variant="bodyStrong">
+              {legend}&nbsp;
+              {required ? (
+                <StyledRequiredIcon name="asterisk" color="danger" size={8} />
+              ) : null}
+            </Text>
```
Comment (matthprost) [inline packages/ui/src/components/SelectableCardGroup/index.tsx:145]: Here you can add a condition to check if legend exists or not and do not render it if it doesn't

File: packages/ui/src/components/SelectableCardGroup/index.tsx [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+/**
+ * SelectableCardGroup is a component that allows users to select cards from a list of cards using SelectableCard.
+ */
+export const SelectableCardGroup = ({
+  legend,
+  value,
+  className,
+  helper,
+  error,
+  columns = 1,
```
Comment (matthprost) [inline packages/ui/src/components/SelectableCardGroup/index.tsx:115]: That's a really nice idea I like it 🚀

**Review by matthprost — CHANGES_REQUESTED** [review]
I added some small feedback but overall you did an excellent job congrats 😄

**Pull request merged**